### PR TITLE
[EXEC] Support fcompute

### PR DESCRIPTION
**Pull request by tqchen**
@piiswrong

### CODE REVIEW

File: src/executor/attach_op_execs_pass.cc [hunk 1]
```diff
@@ -159,6 +203,8 @@ Graph AttachOpExecs(Graph g) {
           dynamic_cast<ForwardOpExecutor*>(ret[fwd_id].get())->op_,
           mxnet::op::OpPropGetOpProperty(inode.source->attrs),
           mutate_index);
+    } else if (fcompute != nullptr) {
```
Comment (piiswrong) [inline src/executor/attach_op_execs_pass.cc]: should fcompute take priority over create layer?

File: src/executor/attach_op_execs_pass.cc [hunk 2]
```diff
@@ -140,6 +183,7 @@ Graph AttachOpExecs(Graph g) {
     if (fmutate_inputs.count(inode.source->op())) {
       mutate_index = fmutate_inputs[inode.source->op()](inode.source->attrs);
     }
+    FCompute fcompute = FComputeExecutor::GetFCompute(inode.source->op(), vctx[i]);
```
Comment (piiswrong) [inline src/executor/attach_op_execs_pass.cc]: why not use Executor::Run directly?

File: src/executor/attach_op_execs_pass.cc (hunk 1 above)
Comment (tqchen) [inline src/executor/attach_op_execs_pass.cc]: normally it is one or another, so didn't care about priority here

File: src/executor/attach_op_execs_pass.cc (hunk 2 above)
Comment (tqchen) [inline src/executor/attach_op_execs_pass.cc]: don;t understand this comment

**Comment by tqchen** [comment]
merge as lint passed, There is a documentation error that is yet to be fixed by ImperativeInvoke https://travis-ci.org/dmlc/mxnet/jobs/158363420

@piiswrong Please fix it in your next PR

**Pull request merged**